costmap_2d/CMakeLists.txt

```diff
@@ -79,10 +79,6 @@ catkin_package(
         tf
         visualization_msgs
         voxel_grid
-    DEPENDS
```

Comment:
why? this is correct

Reply:
the DEPENDS is not needed. If it is added CLion will complain

Reply:
http://wiki.ros.org/catkin/CMakeLists.txt (point 6 and http://answers.ros.org/question/58498/what-is-the-purpose-of-catkin_depends/ ) DEPENDS is used make sure that external packages are forwarded properly. If you do not have it then all the other packages that use this will need to find package these, which is not nice. No idea why CLion is complaining though, may be its a spelling issue.

Reply:
CMake Error at cmake-build-debug/devel/share/costmap_2d/cmake/costmap_2dConfig.cmake:148 (message):
  Project 'base_local_planner' tried to find library '-lpthread'.  The
  library is neither a target nor built/installed properly.  Did you compile
  project 'costmap_2d'? Did you find_package() it before the subdirectory
  containing its code is included?
Call Stack (most recent call first):
  /opt/ros/kinetic/share/catkin/cmake/catkinConfig.cmake:76 (find_package)
  sootballs/navigation/base_local_planner/CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/common/ws/src/cmake-build-debug/CMakeFiles/CMakeOutput.log".
See also "/home/common/ws/src/cmake-build-debug/CMakeFiles/CMakeError.log".

I don't think its needed, though can we just remove it? Its quite annoying

Reply:
porting to melodic probably it might disappear there

Reply:
lets decline this